sklearn/feature_extraction/hashing.py

```diff
@@ -113,11 +120,26 @@ def _validate_params(n_features, input_type):
             raise ValueError("input_type must be 'dict', 'pair' or 'string',"
                              " got %r." % input_type)
 
-    def fit(self, X=None, y=None):
-        """No-op.
+    def _learn_params(self, raw_X):
```

Comment:
So I've moved the learning logic outside and calling this from `fit` if `save_mappings` is True. So your request for learning the mappings at the `fit` time is achieved. If `save_mappings` is False, then the values are being learned at `transform` time,